src/data/move-attrs/sacrificial-full-restore-attr.ts

```diff
@@ -45,8 +51,12 @@ export class SacrificialFullRestoreAttr extends SacrificialAttr {
     return -20;
   }
 
+  /**
+   * Only works if there is at least 1 unfainted allowed Pokemon in the party and not already in battle
+   * @returns `true` if the condition is met, `false` otherwise
+   */
   override getCondition(): MoveConditionFunc {
-    return (_user, _target, _move) =>
-      globalScene.getPlayerParty().filter((p) => p.isActive()).length > globalScene.currentBattle.getBattlerCount();
+    return (user, _target, _move) =>
+      user.getParty().filter((p) => p.isActive()).length > globalScene.currentBattle.getBattlerCount();
```

Comment:
```suggestion
      user.getParty().filter((p) => p.isActive(true)).length > globalScene.currentBattle.getBattlerCount();
```
Accounting for challenges

Reply:
I don't think that's correct, `isActive(true)` returns the active pokemon on the field if I recall correctly
what we need is to make sure that there is at least one active mon that is not on the field

I think the current implementation worked when I playtested it

Reply:
Yes. `isActive` takes in a boolean of whether or not the Pokemon is on the field or not. By default `isActive` has that param as false so it's checking for Pokemon that are unfainted, allowed in the challenge, and not on the field